lib/getDbQuery.js

```diff
@@ -0,0 +1,14 @@
+const getDbQuery = (urlParams = '') => {
+  const query = {}
+  const { q: searchString } = urlParams
+
+  query['shops.0'] = { $exists: true }
+
+  if (searchString) {
+    query['model'] = { $regex: RegExp(`.*${searchString}.*`, 'i') }
+    return query
+  }
+  return query
+}
+
+module.exports = getDbQuery
```

Comment:
This is a very smart solution! You could add error handling, if the query is empty, instead of returning an empty object. But maybe you do this already somewhere else I didn't see.

Reply:
it's not an error if the query is empty. it only means it calls the DB and fetch all sneaker without any additional search condtions